staging/src/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal_test.go

```diff
@@ -98,6 +98,40 @@ func TestEqualities(t *testing.T) {
 	}
 }
 
+func TestPublicEqualities(t *testing.T) {
+	e := Equalities{}
+
+	type Foo struct {
+		X int
+		y int
+	}
+
+	variantA := Foo{1, 2}
```

Comment:
I'm not sure different int values help with coverage.  How about testing different compositions?  Fields containing structs?  Embeddedd structs? Arrays of structs, maps with struct values?  Unexported structs?  ...